src/csvReaderWrapper.js

```diff
@@ -1,30 +1,38 @@
 import Papa from "papaparse";
 import { store } from "./index.js";
-import { setImportedData, setColumnsByDataType } from "./redux";
+import {
+  setImportedData,
+  setColumnsByDataType,
+  columnContainsOnlyNumbers
+} from "./redux";
 import { ColumnTypes } from "./constants.js";
 
-export const parseCSV = (csvfile, download, setColumnsToOther) => {
+export const parseCSV = (csvfile, download, useDefaultColumnDataType) => {
   Papa.parse(csvfile, {
     complete: result => {
-      updateData(result, setColumnsToOther);
+      updateData(result, useDefaultColumnDataType);
     },
     header: true,
     download: download,
     skipEmptyLines: true
   });
 };
 
-const updateData = (result, setColumnsToOther) => {
+const updateData = (result, useDefaultColumnDataType) => {
   var data = result.data;
-
   store.dispatch(setImportedData(data));
-  if (setColumnsToOther) {
+  if (useDefaultColumnDataType) {
     setDefaultColumnDataType(data);
   }
 };
 
 const setDefaultColumnDataType = data => {
-  Object.keys(data[0]).map(column =>
-    store.dispatch(setColumnsByDataType(column, ColumnTypes.CATEGORICAL))
-  );
+  const columns = Object.keys(data[0]);
+  for (let column of columns) {
+    if (columnContainsOnlyNumbers(data, column)) {
+      store.dispatch(setColumnsByDataType(column, ColumnTypes.NUMERICAL))
+    } else {
+      store.dispatch(setColumnsByDataType(column, ColumnTypes.CATEGORICAL))
```

Comment:
Would we want to handle a case where the column values are numbers but it's considered categorical (e.g., `Slope of Peak Exercise ST Segment` column in the Hearts dataset)?

Reply:
I thought about this, but wasn't convinced there was a fool-proof heuristic to auto-detect that case. What do you think? If the column contains only numbers and has less than X unique values? I opted not to do something like that because the more complexity we have behind the scenes the more we have to expose to students so they can reason about what's happening. We don't want to be _too_ clever with our guesses and we'll never be right 100% of the time, which is why we still allow students to change data type in the column inspector dropdown.

Reply:
I think the line we're trying to walk is "helpful, not perfect". There's curricular value in having students think critically about data types, including situations where numbers actually represent categories (like # of legs in the Zoo dataset). I think doing our best to autodetect, but then giving students the option to change via the dropdown, is the best move here - it still gives students a lever of control and ownership over their data, without being so tedious that they need to update every single column (which was a problem in the past)

Reply:
That makes sense! Thank you both for the thoughtful discussion.